Document `Icon` component (#1356)
iTwin/stratakit: apps/website/src/content/docs/components/stratakit/Icon.md

title: Icon
description: Icons are used to visually represent actions or information.
links:
  apiReference: /reference/mui/Icon/
---

::example{src="mui/Icon.default"}

## Examples

### Sizes

Use the `size` prop to adjust the **Icon’s** physical dimensions.

- **Regular:** Default size, suitable for most use cases.
- **Large:** Increased size, use in larger spaces.

::example{src="mui/Icon.sizes"}

:::note[Increase detail]

The [`@stratakit/icons`](https://github.com/iTwin/stratakit/tree/main/packages/icons) package uses [`<symbol>` elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/symbol) to provide multiple resolutions of the same icon. Use a [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment) in the `href` prop to reference a specific symbol. The available symbols are `#icon` (default) and `#icon-large`. In the example above, `#icon-large` symbol is applied to the large icon to display a more detailed icon.

> :::

Author: [Move href prop documentation up](https://github.com/iTwin/stratakit/pull/1356/commits/e5554a8b709152417326ef2581659c6c592e31a7) (and include links to the `/icons` page).

---

I think we need to (separately) update the `/icons` page to document the available fragments and display large icons (maybe a toggle?).

### Decorative

Omit the `alt` prop if the **Icon** is purely decorative. This will hide the **Icon** from assistive technologies.

::example{src="mui/Button.icon"}

### Color

Use the [`color` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/color) to adjust the **Icon’s** color.

::example{src="mui/Icon.color"}

### Custom

Use the `render` prop to display a custom SVG icon.

::example{src="mui/Icon.custom"}

## ✅ Do

- If an accessible name is not provided by other means, use the `alt` prop to provide a descriptive label.
- Increase **Icon** detail when physical size increases.

## 🚫 Don’t

- Don’t use the `alt` prop if there is already an accessible label in place (as in `IconButton`).
- Don’t use custom icons when a standard icon is available in `@stratakit/icons`.
- Don't use non-standard icons from `@mui/icons-material`.

> - Don't use the `SvgIcon` component from `@mui/material`.

Author: https://github.com/iTwin/stratakit/pull/1341#discussion_r3045713856

title: Icon
description: Icons are used to visually represent actions or information.
links:
  apiReference: /reference/mui/Icon/
---

::example{src="mui/Icon.default"}

> The **Icon** component ensures a consistent visual style for icons throughout the user interface. It is optimized for use with the [icons](/icons/) from the [`@stratakit/icons`](https://github.com/iTwin/stratakit/tree/main/packages/icons) package, but it also supports [custom icons](#custom).

Reviewer: We're already linking to the package in the `/icons` page and in the paragraph right after.

```suggestion
The **Icon** component ensures a consistent visual style for icons throughout the user interface. It is optimized for use with [StrataKit icons](/icons/), but it also supports [custom icons](#custom).
```

> Use the `href` prop to specify the icon you want to display. This prop should be set to a URL to an `.svg` from the [`@stratakit/icons`](https://www.npmjs.com/package/@stratakit/icons?activeTab=versions) package, or a [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment) pointing to a [`<symbol>` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/symbol) inside an SVG. All StrataKit icons include two symbols with different levels of detail: `#icon` (the default) and `#icon-large`.

Reviewer: ```suggestion
Use the [`href`](/reference/mui/Icon/#Icon.Icon.href) prop to specify the icon you want to display. This prop should be set to a URL to an `.svg` from the [`@stratakit/icons`](https://www.npmjs.com/package/@stratakit/icons?activeTab=versions) package, or a [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment) pointing to a [`<symbol>` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/symbol) inside an SVG. All StrataKit icons include two symbols with different levels of detail: `#icon` (the default) and `#icon-large`.
```

## Examples

### Sizes

> Use the `size` prop to adjust the **Icon's** physical dimensions.

Reviewer: ```suggestion
Use the [`size`](/reference/mui/Icon/#Icon.Icon.size) prop to adjust the **Icon's** physical dimensions.
```

> Use the `href` prop to specify the URL of the icon you want to display. The [icons](/icons/) from the [`@stratakit/icons`](https://github.com/iTwin/stratakit/tree/main/packages/icons) package use [`<symbol>` elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/symbol), which allow for different levels of icon detail. You can specify the desired detail level by including a [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment) in the `href`. The available symbols are `#icon` (the default) and `#icon-large`.

Reviewer: Better clarity on:
- The values accepted by the `href` prop.
- The relationship between the URI fragment and the symbol elements.

```suggestion
Use the `href` prop to specify the icon you want to display. This prop should be set to a URL to an `.svg` from the [`@stratakit/icons`](https://www.npmjs.com/package/@stratakit/icons?activeTab=versions) package, or a [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment) pointing to a [`<symbol>` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/symbol) inside an SVG. All StrataKit icons include two symbols with different levels of detail: `#icon` (the default) and `#icon-large`.
```

- **Regular:** Default size, suitable for most use cases.
- **Large:** Increased size, use in larger spaces.

::example{src="mui/Icon.sizes"}

:::note[Increase detail]

> In the example above, `#icon-large` symbol is applied to the large icon to display a more detailed icon.

Reviewer: ```suggestion
In the example above, the `#icon-large` symbol is applied to the large icon to display a more detailed icon.
```

:::

### Decorative

> Omit the `alt` prop if the **Icon** is purely decorative. This will hide the **Icon** from assistive technologies.

Reviewer: Should explain why this example is decorative.

```suggestion
Omit the [`alt`](/reference/mui/Icon/#Icon.Icon.alt) prop if the **Icon** is purely decorative. This will hide the **Icon** from assistive technologies.

In the example below, the Button already has an accessible name, so the **Icon** is considered decorative.
```

::example{src="mui/Button.icon"}

### Color

Use the [`color` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/color) to adjust the **Icon's** color. Prefer usage of StrataKit color tokens when adjusting the **Icon's** color.

::example{src="mui/Icon.color"}

### Custom

> Use the `render` prop to display a custom SVG icon.

Reviewer: We should also document how to display a custom external `.svg`. At least in text, if not a full example.

Author: [Custom Icon href](https://github.com/iTwin/stratakit/pull/1356/commits/7810014a0492b9dec65de69bf553fbba660e2d61)

> Use the `render` prop to display a custom inline SVG icon. Alternatively, you can use the `href` prop to reference a custom icon or a specific `<symbol>` element within an external SVG file.

Reviewer: We should flip the order of these two methods and add some guidance.

```suggestion
The `href` prop can point to any valid `<symbol>` or `<svg>` that has an `id`. This is the recommended way to use custom icons.

Alternatively, the `render` prop can be used to display an inlined SVG, as in the example below. This can be useful when you need to target the individual elements inside the SVG (e.g. for styling or animation purposes).
```

title: Icon
description: Icons are used to visually represent actions or information.

> links:

Reviewer: All custom components should have a status.

```suggestion
status: stable
links:
```

links:

>   apiReference: ../../reference/foundations/Icon/

Reviewer: Absolute path.

```suggestion
  apiReference: /reference/foundations/Icon/
```

It'd be good to also mention somewhere that this component is exported from both `@stratakit/mui` and `@stratakit/foundations`.
